policy/file/source.go

```diff
@@ -266,8 +270,13 @@ func (s *Source) handleDir() ([]policy.PolicyID, error) {
 
 		scalingPolicy.ApplyDefaults(s.config)
 
+		if err := scalingPolicy.Validate(); err != nil {
+			mErr = multierror.Append(fmt.Errorf("failed to validate file %s: %v", file, err), mErr)
+			continue
+		}
+
 		for _, c := range scalingPolicy.Checks {
-			c.CanonicalizeAPMQuery(scalingPolicy.Target)
+			c.Canonicalize(scalingPolicy.Target)
 		}
```

Comment:
We could also have a `func (p *Policy) Canonicalize()` that does this loop to canonicalize the checks.